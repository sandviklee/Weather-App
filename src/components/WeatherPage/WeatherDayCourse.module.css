.main {
    position: relative;
    margin: auto;
    height: auto;
    width: 48vw;
    border: 0px;
    border-top: 3px;
    border-bottom: 3px;
    border-color: black;
    border-style: solid;
    padding-top: 20px;
    padding-bottom: 20px;
}

.extremalTemp {
    font-family: "Inter", sans-serif;
    font-weight: 600;
    font-size: 20px;
    margin-top: 1vh;
}

.temperatures {
    margin-top: -1vh;
}

.weathers {
    position: relative;
    display: flex;
    column-gap: 1px;
}

.container {
    margin: 0 auto;
    align-items: center;
    display: grid;
    grid-auto-flow: row;
    grid-row-gap: 2px;
}

.container p {
    margin: 0;
    margin-top: 30px;
    font-family: "Inter", sans-serif;
    padding-left: 4px;
    font-size: 20px;
}

.container sub{
    font-size: 18px;
}

.weatherImage {
    width: 6vh;
}

.icon {
    position: relative;
    font-size: 30px;
    bottom: -0.5vh;
}

@media only screen and (max-width: 1000px) {
    .extremalTemp {
        font-size: 2vw;
    }
    
    .temperatures {
        font-size: 2vw;
    }

    .container p {
        font-size: 2vw;
    }
}

@media only screen and (max-width: 600px) {
    .main {
        width: 80vw;
    }
    .extremalTemp {
        font-size: 4vw;
    }
    
    .temperatures {
        font-size: 4vw;
    }

    .container p {
        font-size: 4vw;
    }
}

@media only screen and (max-height: 800px) {
    .weatherImage {
        width: 6vw;
    }
}