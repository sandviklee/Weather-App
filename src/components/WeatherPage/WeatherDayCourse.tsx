import style from "./WeatherDayCourse.module.css";

interface DayCourseProps {
    extremalTemperatures: Array<string>;
    weatherCourse: Array<string>;
}

/**
 * This is the Course of day Weather Information component.
 * @summary Renders a component with information about the course of day
 * @param props (extrmalTemperatures, weatherCourse)
 * @returns Weather Course of Day JSX Element
 */
const WeatherDayCourse = (props: DayCourseProps): JSX.Element => {
    return (
        <main className={style.main}>
            <p className={style.extremalTemp}>Maks/Min Temperatur</p>
            <p className={style.temperatures}>
                {props.extremalTemperatures[0]}C° /{" "}
                {props.extremalTemperatures[1]}C°
            </p>

            <div className={style.weathers}>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>NATT</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[0]}.svg`}
                            alt="Morning Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>MORGEN</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[1]}.svg`}
                            alt="Noon Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>AFTEN</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[2]}.svg`}
                            alt="Afternoon Weather"
                        />
                    </div>
                </div>
                <div className={style.container}>
                    <div className={style.weatherContainer}>
                        <p>KVELD</p>
                        <img
                            className={style.weatherImage}
                            src={`../../src/assets/images/svg/${props.weatherCourse[3]}.svg`}
                            alt="Night Weather"
                        />
                    </div>
                </div>
            </div>
        </main>
    );
};

export default WeatherDayCourse;
