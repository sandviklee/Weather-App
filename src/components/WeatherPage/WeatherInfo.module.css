.main {
    position: relative;
    display: flex;
    margin: auto;
    border-radius: 20px;
    height: auto;
    padding-top: 3vh;
    padding-bottom: 4vh;
    width: 48vw;
    min-width: 40%;
}

.container {
    margin: 0 auto;
    display: flex;
    align-items: center;
}

.container p {
    padding-left: 4px;
    font-size: 50px;
    margin: 0;
}

.container sub{
    font-size: 18px;
}

.icon {
    position: relative;
    font-size: 30px;
    bottom: -0.5vh;
}

@media only screen and (max-width: 1000px) {
    .main {
        flex-wrap: wrap;
        justify-content: center;
        padding-top: 1vw;
    }
    .container {
        margin: 0 2vw 0 2vw;
        max-height: 12vw;
        padding: 1vw 0 1vw 0;
    }
    .container p {
        font-size: 6vw;
    }
    .icon {
        font-size: 5vw;
    }
}
@media only screen and (max-width: 600px) {
    .main {
        width: 80vw;
        flex-wrap: wrap;
        justify-content: center;
    }

    .container {
        margin: 0 2vw 0 2vw;
        max-height: 12vw;
    }

    .container p {
        font-size: 8vw;
    }
    .icon {
        font-size: 7vw;
    }
}