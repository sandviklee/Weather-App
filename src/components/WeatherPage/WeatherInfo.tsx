import style from "./WeatherInfo.module.css";
import { BsThermometerHalf, BsUmbrella, BsWind } from "react-icons/bs";

interface WeatherProps {
    temperature: string;
    rain: string;
    wind: string;
}

const WeatherInfo = (props: WeatherProps) => {
    return (
        <main className={style.main}>
            <div className={style.container}>
                <BsThermometerHalf className={style.icon} />
                <p className={style.information}>{props.temperature} CÂ°</p>
            </div>
            <div className={style.container}>
                <BsUmbrella className={style.icon} />
                <p className={style.information}>
                    {props.rain} <sub>mm</sub>
                </p>
            </div>
            <div className={style.container}>
                <BsWind className={style.icon} />
                <p className={style.information}>
                    {props.wind} <sub>m/s</sub>
                </p>
            </div>
        </main>
    );
};

export default WeatherInfo;
