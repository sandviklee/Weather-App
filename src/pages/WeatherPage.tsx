import { useQuery } from "@tanstack/react-query";
import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import style from "../style/Weather.module.css";
import WeatherInfo from "../components/WeatherPage/WeatherInfo";
import WeatherDayCourse from "../components/WeatherPage/WeatherDayCourse";
import TodaysDate from "../components/Date/TodaysDate";
import TodaysDateComponent from "../components/Date/TodaysDateComponent";
import { AiOutlineHeart, AiFillCalendar } from "react-icons/ai";

interface WeatherProps {
    lat: string;
    lon: string;
}
/**
 * This is the Weather page component which is depends on longitude and latitude.
 * @summary Will render a page that shows weather information from todays date.
 * @param props (latitude and longitude)
 * @returns Weather Page JSX Element
 */
const WeatherPage = (props: WeatherProps): JSX.Element => {
    const { id } = useParams();
    const [weather, setWeather] = useState(null);
    const [temperature, setTemperature] = useState("0");
    const [precipitation, setPrecipitation] = useState("0");
    const [wind, setWind] = useState("0");
    const [extremalTemp, setExtremalTemp] = useState(Array("0", "0"));
    const [dayCourse, setDayCourse] = useState(
        Array("no_weather", "no_weather", "no_weather", "no_weather")
    );

    const endpoint: string = `https://api.met.no/weatherapi/locationforecast/2.0/complete?lat=${props.lat}&lon=${props.lon}`;
    const time: string = new Date().toJSON().split("T")[1];

    const { data } = useQuery({
        /**
         * @summary TanStack Query, fetches API from MET to data
         */
        queryKey: [""],
        queryFn: () => fetch(endpoint).then((res) => res.json()),
    });

    const filterTodaysWeatherData: any = (data: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Filtered weather data based on todays date
         */
        return data["properties"]["timeseries"].filter(
            (value: { [x: string]: string }) =>
                value["time"].split("T")[0] == TodaysDate()
        );
    };

    const sortWeatherByTemperature = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Filtered weather data based on highest to lowest temperature
         */
        let newWeather = Object.create(weather);
        return newWeather.sort((a: any, b: any) => {
            return (
                parseFloat(b["data"]["instant"]["details"]["air_temperature"]) -
                parseFloat(a["data"]["instant"]["details"]["air_temperature"])
            );
        });
    };

    const filterWeatherToCurrentTime: any = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Filtered weather data based on the current hour
         */
        return weather.filter(
            (value: { [x: string]: any }) =>
                value["time"].split("T")[1].slice(0, -1).split(":")[0] ==
                time.split(":")[0]
        );
    };

    const getCurrentTemperature = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Temperature data at current hour
         */
        return filterWeatherToCurrentTime(weather)[0]["data"]["instant"][
            "details"
        ]["air_temperature"];
    };

    const getPrecipitation1H = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Precipitation data over 1 hour from current hour
         */
        return filterWeatherToCurrentTime(weather)[0]["data"]["next_1_hours"][
            "details"
        ]["precipitation_amount"];
    };

    const getCurrentWind = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Wind data from current hour
         */
        return filterWeatherToCurrentTime(weather)[0]["data"]["instant"][
            "details"
        ]["wind_speed"];
    };

    const getExtremalTemperatures = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Both Max and Min temperatures from current day
         */
        return new Array(
            sortWeatherByTemperature(weather)[0]["data"]["instant"]["details"][
                "air_temperature"
            ],
            sortWeatherByTemperature(weather)[weather.length - 1]["data"][
                "instant"
            ]["details"]["air_temperature"]
        );
    };

    const getWeatherSymbol = (weather: any, hour: number) => {
        /**
         * @param weather Weather data from MET
         * @param hour    Given hour to get symbol from
         * @returns       Symbol from given hour 
         */
        return weather[hour]["data"]["next_1_hours"]["summary"]["symbol_code"];
    };

    const getDayCourse = (weather: any) => {
        /**
         * @param weather Weather data from MET
         * @returns       Array of the course of symbols from current day
         */
        return new Array(
            weather.length > 17 ? getWeatherSymbol(weather, 6) : "no_weather",
            weather.length > 13
                ? getWeatherSymbol(weather, 10 - (23 - weather.length))
                : "no_weather",
            weather.length > 7
                ? getWeatherSymbol(weather, 16 - (23 - weather.length))
                : "no_weather",
            weather.length > 2
                ? getWeatherSymbol(weather, 21 - (23 - weather.length))
                : "no_weather"
        );
    };

    useEffect(() => {
        if (data == null) {
            return;
        }
        setWeather(filterTodaysWeatherData(data));
    }, [data]);

    useEffect(() => {
        if (weather == null) {
            return;
        }
        setTemperature(getCurrentTemperature(weather));
        setPrecipitation(getPrecipitation1H(weather));
        setWind(getCurrentWind(weather));
        setExtremalTemp(getExtremalTemperatures(weather));
        setDayCourse(getDayCourse(weather));
    }, [weather]);

    return (
        <main className={style.main}>
            <div className={style.container}>
                <div className={style.titleContent}>
                    <div className={style.titlePlacement}>
                        <p className={style.title}>{id?.toUpperCase()}</p>
                        <button className={style.heartButton}>
                            <AiOutlineHeart className={style.heart} />
                        </button>
                    </div>
                    <p className={style.cityTitle}>by (Tr√∏ndelag)</p>
                </div>
                <hr
                    style={{
                        width: "48%",
                        marginTop: "10px",
                        marginBottom: "2px",
                    }}
                />
                <div className={style.contentBackground} />
                <div className={style.content}>
                    <div className={style.date}>
                        <AiFillCalendar />
                        <TodaysDateComponent />
                    </div>
                    <WeatherInfo
                        temperature={temperature}
                        rain={precipitation}
                        wind={wind}
                    />
                    <WeatherDayCourse
                        extremalTemperatures={extremalTemp}
                        weatherCourse={dayCourse}
                    />
                </div>
            </div>
        </main>
    );
};

export default WeatherPage;
