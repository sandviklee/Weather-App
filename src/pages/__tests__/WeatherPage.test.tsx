import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter, Route, Routes} from 'react-router-dom';
import WeatherPage from '../WeatherPage';

// Create a client for React Query
const queryClient = new QueryClient();

describe('WeatherPage component', () => {
    test('renders WeatherPage component', () => {
        render(
            <QueryClientProvider client={queryClient}>
                <MemoryRouter initialEntries={['/weather/Oslo/60/10']}>
                    <Routes>
                        <Route path="/weather/:title/:lat/:lon" element={<WeatherPage />} />
                    </Routes>
                </MemoryRouter>
            </QueryClientProvider>
        );

        // Check if the title is rendered
        expect(screen.getByText('OSLO')).toBeInTheDocument();
        expect(screen.getByText('0 CÂ°')).toBeInTheDocument();
        expect(screen.getByText('Maks/Min Temperatur')).toBeInTheDocument();
        expect(screen.getByText('NATT')).toBeInTheDocument();
        expect(screen.getByText('KVELD')).toBeInTheDocument();   
    });

    test("matches snapshot", () => {
        const { asFragment } = render(
            <QueryClientProvider client={queryClient}>
                <MemoryRouter initialEntries={['/weather/Oslo/60/10']}>
                    <Routes>
                        <Route path="/weather/:title/:lat/:lon" element={<WeatherPage />} />
                    </Routes>
                </MemoryRouter>
            </QueryClientProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});