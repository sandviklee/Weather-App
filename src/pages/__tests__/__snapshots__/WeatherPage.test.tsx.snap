// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherPage component > matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="_main_8ffdce"
  >
    <div
      class="_container_8ffdce"
    >
      <div
        class="_titleContent_8ffdce"
      >
        <div
          class="_titlePlacement_8ffdce"
        >
          <a
            href="/"
          >
            <svg
              class="_arrow_8ffdce"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </a>
          <p
            class="_title_8ffdce"
          >
            OSLO
          </p>
          <button
            class="_heartButton_8ffdce"
          >
            <svg
              class="_heart_8ffdce"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: gray;"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="_contentBackground_8ffdce"
      />
      <div
        class="_content_8ffdce"
      >
        <div
          class="_date_8ffdce"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M112 880c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V460H112v420zm768-696H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v176h800V216c0-17.7-14.3-32-32-32z"
            />
          </svg>
          <p>
            Søndag 26. Nov
          </p>
        </div>
        <main
          class="_main_0824fb"
        >
          <div
            class="_container_0824fb"
          >
            <svg
              class="_icon_0824fb"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.5 12.5a1.5 1.5 0 1 1-2-1.415V6.5a.5.5 0 0 1 1 0v4.585a1.5 1.5 0 0 1 1 1.415z"
              />
              <path
                d="M5.5 2.5a2.5 2.5 0 0 1 5 0v7.55a3.5 3.5 0 1 1-5 0V2.5zM8 1a1.5 1.5 0 0 0-1.5 1.5v7.987l-.167.15a2.5 2.5 0 1 0 3.333 0l-.166-.15V2.5A1.5 1.5 0 0 0 8 1z"
              />
            </svg>
            <p
              class="_information_0824fb"
            >
              0 C°
            </p>
          </div>
          <div
            class="_container_0824fb"
          >
            <svg
              class="_icon_0824fb"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a.5.5 0 0 1 .5.5v.514C12.625 1.238 16 4.22 16 8c0 0 0 .5-.5.5-.149 0-.352-.145-.352-.145l-.004-.004-.025-.023a3.484 3.484 0 0 0-.555-.394A3.166 3.166 0 0 0 13 7.5c-.638 0-1.178.213-1.564.434a3.484 3.484 0 0 0-.555.394l-.025.023-.003.003s-.204.146-.353.146-.352-.145-.352-.145l-.004-.004-.025-.023a3.484 3.484 0 0 0-.555-.394 3.3 3.3 0 0 0-1.064-.39V13.5H8h.5v.039l-.005.083a2.958 2.958 0 0 1-.298 1.102 2.257 2.257 0 0 1-.763.88C7.06 15.851 6.587 16 6 16s-1.061-.148-1.434-.396a2.255 2.255 0 0 1-.763-.88 2.958 2.958 0 0 1-.302-1.185v-.025l-.001-.009v-.003s0-.002.5-.002h-.5V13a.5.5 0 0 1 1 0v.506l.003.044a1.958 1.958 0 0 0 .195.726c.095.191.23.367.423.495.19.127.466.229.879.229s.689-.102.879-.229c.193-.128.328-.304.424-.495a1.958 1.958 0 0 0 .197-.77V7.544a3.3 3.3 0 0 0-1.064.39 3.482 3.482 0 0 0-.58.417l-.004.004S5.65 8.5 5.5 8.5c-.149 0-.352-.145-.352-.145l-.004-.004a3.482 3.482 0 0 0-.58-.417A3.166 3.166 0 0 0 3 7.5c-.638 0-1.177.213-1.564.434a3.482 3.482 0 0 0-.58.417l-.004.004S.65 8.5.5 8.5C0 8.5 0 8 0 8c0-3.78 3.375-6.762 7.5-6.986V.5A.5.5 0 0 1 8 0zM6.577 2.123c-2.833.5-4.99 2.458-5.474 4.854A4.124 4.124 0 0 1 3 6.5c.806 0 1.48.25 1.962.511a9.706 9.706 0 0 1 .344-2.358c.242-.868.64-1.765 1.271-2.53zm-.615 4.93A4.16 4.16 0 0 1 8 6.5a4.16 4.16 0 0 1 2.038.553 8.688 8.688 0 0 0-.307-2.13C9.434 3.858 8.898 2.83 8 2.117c-.898.712-1.434 1.74-1.731 2.804a8.687 8.687 0 0 0-.307 2.131zm3.46-4.93c.631.765 1.03 1.662 1.272 2.53.233.833.328 1.66.344 2.358A4.14 4.14 0 0 1 13 6.5c.77 0 1.42.23 1.897.477-.484-2.396-2.641-4.355-5.474-4.854z"
              />
            </svg>
            <p
              class="_information_0824fb"
            >
              0 
              <sub>
                mm
              </sub>
            </p>
          </div>
          <div
            class="_container_0824fb"
          >
            <svg
              class="_icon_0824fb"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.5 2A2.5 2.5 0 0 0 10 4.5a.5.5 0 0 1-1 0A3.5 3.5 0 1 1 12.5 8H.5a.5.5 0 0 1 0-1h12a2.5 2.5 0 0 0 0-5zm-7 1a1 1 0 0 0-1 1 .5.5 0 0 1-1 0 2 2 0 1 1 2 2h-5a.5.5 0 0 1 0-1h5a1 1 0 0 0 0-2zM0 9.5A.5.5 0 0 1 .5 9h10.042a3 3 0 1 1-3 3 .5.5 0 0 1 1 0 2 2 0 1 0 2-2H.5a.5.5 0 0 1-.5-.5z"
              />
            </svg>
            <p
              class="_information_0824fb"
            >
              0 
              <sub>
                m/s
              </sub>
            </p>
          </div>
        </main>
        <main
          class="_main_26373e"
        >
          <p
            class="_extremalTemp_26373e"
          >
            Maks/Min Temperatur
          </p>
          <p
            class="_temperatures_26373e"
          >
            0C° / 0C°
          </p>
          <div
            class="_weathers_26373e"
          >
            <div
              class="_container_26373e"
            >
              <div
                class="_weatherContainer_26373e"
              >
                <p>
                  NATT
                </p>
                <img
                  alt="Morning Weather"
                  class="_weatherImage_26373e"
                  src="/project1/yr-icons/no_weather.svg"
                />
              </div>
            </div>
            <div
              class="_container_26373e"
            >
              <div
                class="_weatherContainer_26373e"
              >
                <p>
                  MORGEN
                </p>
                <img
                  alt="Noon Weather"
                  class="_weatherImage_26373e"
                  src="/project1/yr-icons/no_weather.svg"
                />
              </div>
            </div>
            <div
              class="_container_26373e"
            >
              <div
                class="_weatherContainer_26373e"
              >
                <p>
                  AFTEN
                </p>
                <img
                  alt="Afternoon Weather"
                  class="_weatherImage_26373e"
                  src="/project1/yr-icons/no_weather.svg"
                />
              </div>
            </div>
            <div
              class="_container_26373e"
            >
              <div
                class="_weatherContainer_26373e"
              >
                <p>
                  KVELD
                </p>
                <img
                  alt="Night Weather"
                  class="_weatherImage_26373e"
                  src="/project1/yr-icons/no_weather.svg"
                />
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
